### PR TITLE
SLM-66: Prevent new barcodes being created by re-printing the PDF

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -60,6 +60,10 @@ services:
     container_name: gotenberg
     ports:
       - "3001:3000"
+    environment:
```
Comment (nathanrussell-moj-digital) [inline docker-compose.yml:63]: I was getting timeout issues with gotenberg in my local docker compose env - given that we set these exact same env values in helm's `values.yml` I thought we'd better do it locally as well!

File: integration_tests/integration/barcode/selectEnvelopeSize.spec.ts [hunk 2]
```diff
@@ -15,6 +15,8 @@ context('Select envelope size', () => {
   })
 
   it('should goto print coversheets given envelope size selected', () => {
+    cy.task('stubCreateBarcode')
```
Comment (nathanrussell-moj-digital) [inline integration_tests/integration/barcode/selectEnvelopeSize.spec.ts:18]: The barcode number is now generated slightly earlier in our process, so need this stub in this test

File: server/@types/prisonTypes.ts [hunk 3]
```diff
@@ -17,4 +17,5 @@ export type Recipient = {
   prisonNumber: string
   prisonerName: string
   prisonAddress: PrisonAddress
+  barcodeValue?: string
```
Comment (nathanrussell-moj-digital) [inline server/@types/prisonTypes.ts:20]: The basic idea is that once we have been through adding all the recipients, at the point of needed barcode numbers (ie. for the copy & paste route, and the PDF route), we have a service call (see below) that iterates through the recipients and calls the API for each one to give them a barcode number. Therefore the collection of recipients in the session will have the details of the prisoner and their prison, plus their barcode number

File: server/services/barcode/CreateBarcodeService.ts [hunk 4]
```diff
@@ -42,28 +36,37 @@ export default class CreateBarcodeService {
...
-        barcode: barcodeValue,
-        barcodeImageDataUrl: this.generateAddressAndBarcodeImage(barcodeImage, recipient),
-      }
-    } catch (error) {
-      logger.error('Error generating barcode image', error)
-      throw error
-    }
-  }
-
-  private async createBarcode(token: string): Promise<string> {
```
Comment (nathanrussell-moj-digital) [inline server/services/barcode/CreateBarcodeService.ts:59]: The `CreateBarcodeService` class now has 3 public methods on its API:
* generateBarcodeValue
* generateAddressAndBarcodeDataUrlImage
* addBarcodeValuesToRecipients

I'm not entirely convinced that addBarcodeValuesToRecipients lives in this class, but given what it does and how it does it I couldnt think of a neater/easier place to put it.

Comment (mikehalmamoj) [reply server/services/barcode/CreateBarcodeService.ts:59]: This class already knows what a Recipient is so I think it's fine

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**